oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -278,11 +292,29 @@ impl Runtime {
     fn drop_abi_handle(&self, node_id: NodeId, handle: oak_abi::Handle) -> Result<(), OakStatus> {
         let mut node_infos = self.node_infos.write().unwrap();
         let node_info = node_infos.get_mut(&node_id).expect("Invalid node_id");
-        node_info
+
+        let event_details = HandleDestroyed {
+            node_id: node_id.0,
+            handle,
+            channel_id: node_info
+                .abi_handles
+                .get(&handle)
+                .ok_or(OakStatus::ErrBadHandle)?
+                .get_id(),
+        };
+
+        let result = node_info
             .abi_handles
             .remove(&handle)
             .ok_or(OakStatus::ErrBadHandle)
-            .map(|_half| ())
+            .map(|_half| ());
+
+        match result {
```

Comment:
nit: am I not sure what is best in terms of idiomatic Rust, but I would have thought a standard `if result.is_ok()` is easier to read.

Reply:
Makes sense, happy to use that instead. To be entirely honest I largely used a match since I didn't know the `is_ok()` method existed yet. ^^